packages/react-components/src/components/Picker/Picker.tsx

```diff
@@ -0,0 +1,160 @@
+import * as React from 'react';
+import { Error } from '@livechat/design-system-icons/react/material';
+import { Trigger, TriggerSize } from './Trigger';
+import { IPickerListItem, PickerList } from './PickerList';
+import { Icon, IconSizeName, IconTypeName } from '../Icon';
+import styles from './Picker.module.scss';
+import cx from 'clsx';
+import { KeyCodes } from '../../utils/keyCodes';
+
+const baseClass = 'picker';
+
+export interface IPickerProps {
+  disabled?: boolean;
+  label?: string;
+  error?: string;
+  options: IPickerListItem[];
+  size?: TriggerSize;
+  placeholder?: string;
+  isRequired?: boolean;
+  noSearchResultText?: string;
+  onSelect: (selectedItem: IPickerListItem | null) => void;
+}
+
+export const Picker: React.FC<IPickerProps> = ({
+  disabled,
+  error,
+  label,
+  options,
+  size = 'medium',
+  placeholder = 'Select option',
+  isRequired,
+  noSearchResultText = 'No results found',
+  onSelect,
+}) => {
+  const [isListOpen, setIsListOpen] = React.useState<boolean>(false);
+  const [selectedItem, setSelectedItem] =
+    React.useState<IPickerListItem | null>(null);
+  const [searchPhrase, setSearchPhrase] = React.useState<string | null>(null);
+  const triggerRef = React.useRef<HTMLDivElement>(null);
+
+  React.useEffect(() => {
+    const onDocumentClick = (e: MouseEvent) => {
+      if ((e.target as Element).contains(triggerRef.current) && isListOpen) {
+        return setIsListOpen(false);
+      }
+
+      return;
+    };
+
+    const onKeyDown = (e: KeyboardEvent) => {
+      if (isListOpen && e.key === KeyCodes.tab) {
+        return setIsListOpen(false);
+      }
+
+      return;
+    };
+
+    if (!isListOpen) {
+      setSearchPhrase(null);
+    }
+
+    document.addEventListener('click', onDocumentClick);
+    document.addEventListener('keydown', onKeyDown);
+
+    return () => {
+      document.removeEventListener('click', onDocumentClick);
+      document.addEventListener('keydown', onKeyDown);
+    };
+  }, [isListOpen]);
+
+  const handleOnTriggerClick = () => {
+    if (disabled) {
+      return;
+    }
+
+    return setIsListOpen((prev) => !prev);
+  };
+
+  const handleOnClose = () => {
+    setIsListOpen(false);
+  };
+
+  const handleOnSelect = (item: IPickerListItem) => {
+    setIsListOpen(false);
+    setSelectedItem(item);
+    onSelect(item);
+  };
+
+  const handleOnClearClick = () => {
+    setIsListOpen(false);
+    setSelectedItem(null);
+    onSelect(null);
+  };
+
+  const handleOnFilter = (text: string) => setSearchPhrase(text);
+
+  const getOptions = (): IPickerListItem[] => {
+    if (!searchPhrase) {
+      return options;
+    }
+
+    return options.filter((item: IPickerListItem) => {
+      const search = searchPhrase.toLowerCase();
+      const itemName = item.name.toLowerCase();
+
+      if (itemName.includes(search) && !item.groupHeader) {
+        return item;
+      }
+    });
+  };
```

Comment:
there is more than `searchPhrase` tha can change in this component during its lifecycle, which means `getOptions`Twold be called multiple times unnecessary. Considering that it is doing list filtering, it may be a performance issue for larger lists so it's a good candidate for being a derived and memorized value that changes only when `searchPhrase` has changed. 

Additionally, [`Array.prototype.filter`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) takes a predicated as its argument. A predicate is a function that should always return a boolean value. Here you are returning a specific item element. It works because the item object is cast to boolean type and it is true but it is unnecessary and hard to read. In case of any predicate you can return condition value directly 🙂 

```suggestion
  const items = React.useMemo<IPickerListItem[]>(() => {
    if (!searchPhrase) {
      return options;
    }

    return options.filter((item) => {
      if(item.groupHeader) {
        return false
      }

      const search = searchPhrase.toLowerCase();
      const itemName = item.name.toLowerCase();
      return itemName.includes(search)
    });
  }, [searchPhrase])
```